invokeai/backend/stable_diffusion/diffusion_backend.py

```diff
@@ -122,21 +107,31 @@ def run_unet(self, ctx: DenoiseContext, ext_manager: ExtensionsManager, conditio
         ctx.conditioning_mode = conditioning_mode
         ctx.inputs.conditioning_data.to_unet_kwargs(ctx.unet_kwargs, ctx.conditioning_mode)
 
-        # ext: controlnet/ip/t2i [pre_unet]
-        ext_manager.run_callback(ExtensionCallbackType.PRE_UNET, ctx)
+        # ext: controlnet/ip/t2i [pre_unet_forward]
+        ext_manager.run_callback(ExtensionCallbackType.PRE_UNET_FORWARD, ctx)
 
-        # ext: inpaint [pre_unet, priority=low]
-        # or
-        # ext: inpaint [override: unet_forward]
-        noise_pred = self._unet_forward(**vars(ctx.unet_kwargs))
+        # ext: inpaint model/ic-light [override: unet_forward]
+        noise_pred = ext_manager.run_override(ExtensionOverrideType.UNET_FORWARD, self.unet_forward, ctx, ext_manager)
 
-        ext_manager.run_callback(ExtensionCallbackType.POST_UNET, ctx)
+        ext_manager.run_callback(ExtensionCallbackType.POST_UNET_FORWARD, ctx)
 
         # clean up locals
         ctx.unet_kwargs = None
         ctx.conditioning_mode = None
 
         return noise_pred
 
-    def _unet_forward(self, **kwargs) -> torch.Tensor:
-        return self.unet(**kwargs).sample
+    # pass extensions manager as arg to allow override access it
+    def combine_noise_preds(self, ctx: DenoiseContext, ext_manager: ExtensionsManager) -> torch.Tensor:
+        guidance_scale = ctx.inputs.conditioning_data.guidance_scale
+        if isinstance(guidance_scale, list):
+            guidance_scale = guidance_scale[ctx.step_index]
+
+        # Note: Although this `torch.lerp(...)` line is logically equivalent to the current CFG line, it seems to result
+        # in slightly different outputs. It is suspected that this is caused by small precision differences.
+        # return torch.lerp(ctx.negative_noise_pred, ctx.positive_noise_pred, guidance_scale)
+        return ctx.negative_noise_pred + guidance_scale * (ctx.positive_noise_pred - ctx.negative_noise_pred)
+
+    # pass extensions manager as arg to allow override access it
```

Comment:
How else tiled denoise will be able to call callbacks?

Reply:
Maybe this will be easier to discuss in the context of the tiled denoise PR? It seems to me that if we can avoid passing the ext_manager down to callbacks/overrides then that would keep things quite a bit simpler.